File: project/security/src/main/java/org/tby/fourdk/security/NoSecurityStrategy.java
package org.tby.fourdk.security;

import java.util.List;

public class NoSecurityStrategy implements AuthorizedStrategy {

    private RoleReadRepository roleRepository;

    public NoSecurityStrategy(RoleReadRepository roleRepository) {
        this.roleRepository = roleRepository;
    }

    @Override
    public Authorization isAuthorized(Permission expectedPermission, List<Role> roles) {
        var appPermissions = this.roleRepository.findAllPermissions();
        return new Authorization(true, appPermissions.getAllPermissions());
    }


}

File: rust/project/security-rust/src/dddk/security/authorized_strategy_impl/no_security_strategy.rs
use std::rc::Rc;
use crate::dddk::security::authorization::Authorization;
use crate::dddk::security::authorized_strategy::AuthorizedStrategy;
use crate::dddk::security::authorized_strategy_impl::role_read_repository::RoleReadRepository;
use crate::dddk::security::permission::Permission;

pub struct NoSecurityStrategy {
    role_read_repository: Rc<dyn RoleReadRepository>,
}

impl NoSecurityStrategy {
    pub fn new(role_read_repository: Rc<dyn RoleReadRepository>) -> NoSecurityStrategy {
        NoSecurityStrategy {
            role_read_repository
        }
    }
}

impl AuthorizedStrategy for NoSecurityStrategy{
    fn is_authorized(&self, _expected_permission: Permission, _given_roles: Vec<String>) -> Authorization {
        let permissions = self.role_read_repository.find_all_permissions();
        Authorization::new(true, permissions)
    }
}